tests/recognizer-tests.js

```diff
@@ -129,6 +129,16 @@ test("A dynamic route recognizes", function() {
   equal(router.recognize("/zoo/baz"), null);
 });
 
+test("A dynamic route with encoded params", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo/:bar", handler: handler }]);
+
+  equal(router.recognize("/foo/bar%20baz")[0].params.bar, "bar baz");
+  equal(router.recognize("/foo/bar%23baz")[0].params.bar, "bar#baz");
+  equal(router.recognize("/foo/bar%3fbaz")[0].params.bar, "bar?baz");
+});
+
```

Comment:
Either the paths given to `router.add` should be escaped or not, and same for those given to `router.recognize`. I don't think RouteRecognizer's docs state either way, but the following (already existing) test suggests `add` takes unescaped paths and `recognize` takes escaped paths:
```js
test("A unicode route recognizes", function() {
  var handler = {};
  var router = new RouteRecognizer();
  router.add([{ path: "/uniçø∂∑/ʇɥƃᴉɹlɐ", handler: handler }]);

  var encoded = encodeURI("/uniçø∂∑/ʇɥƃᴉɹlɐ");
  resultsMatch(router.recognize(encoded), [{ handler: handler, params: {}, isDynamic: false }]);
  equal(router.recognize("/uniçø∂∑"), null);
});
```

So it's sensible that I couldn't get a test that assumes (escaped, escaped) to work.